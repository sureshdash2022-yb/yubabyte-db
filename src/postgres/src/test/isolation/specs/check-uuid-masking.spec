setup
{
  CREATE TABLE test (
     k INT PRIMARY KEY,
     v INT
  );

  INSERT INTO test VALUES (1, 1);
}

teardown
{
  DROP TABLE test;
}

session "s1"
setup			{ BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ; }
step "s1_priority"		{ SET yb_transaction_priority_lower_bound = 0.6; }
step "s1_update"		{ UPDATE test SET v=5 where k=1; }
step "s1_commit"		{ COMMIT; }

session "s2"
setup			{ BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ; }
step "s2_priority"		{ SET yb_transaction_priority_upper_bound = 0.4; }
step "s2_update"		{ UPDATE test SET v=6 where k=1; }
step "s2_rollback"		{ ROLLBACK; }

permutation "s1_priority" "s2_priority" "s1_update" "s2_update" "s1_commit" "s2_rollback"
