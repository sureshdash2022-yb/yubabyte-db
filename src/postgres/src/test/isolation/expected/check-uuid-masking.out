Parsed test spec with 2 sessions

starting permutation: s1_priority s2_priority s1_update s2_update s1_commit s2_rollback
step s1_priority: SET yb_transaction_priority_lower_bound = 0.6;
step s2_priority: SET yb_transaction_priority_upper_bound = 0.4;
step s1_update: UPDATE test SET v=5 where k=1;
step s2_update: UPDATE test SET v=6 where k=1;
ERROR:  Operation failed. Try again: ********-****-4***-****-************ Conflicts with higher priority transaction: ********-****-4***-****-************
step s1_commit: COMMIT;
step s2_rollback: ROLLBACK;
